catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/ResourceOperationsOptionsTest.java

```diff
@@ -0,0 +1,256 @@
+        helperGetEnterpriseResourceOptions();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetEnterpriseResourceOptionsCatchesIllegalArgumentException() throws Exception {
+
+        whenQueried().thenThrow(IllegalArgumentException.class);
+        helperGetEnterpriseResourceOptions();
+    }
+
+    @Test
+    public void testGetResourceOptions() throws Exception {
+
+        mockResultList.add(resultMock);
+        whenQueried().thenReturn(queryResponseMock);
+        helperGetResourceOptions();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetResourceOptionsEmptyResults() throws Exception {
+
+        whenQueried().thenReturn(queryResponseMock);
+        helperGetResourceOptions();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetResourceOptionsCatchesUnsupportedQueryException() throws Exception {
+
+        whenQueried().thenThrow(UnsupportedQueryException.class);
+        helperGetResourceOptions();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetResourceOptionsCatchesFederationException() throws Exception {
+
+        whenQueried().thenThrow(FederationException.class);
+        helperGetResourceOptions();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetResourceOptionsCatchesIllegalArgumentException() throws Exception {
+
+        whenQueried().thenThrow(IllegalArgumentException.class);
+        helperGetResourceOptions();
+    }
+
+    @Test
+    public void testGetLocalResourceOptions() throws Exception {
+
+        mockResultList.add(resultMock);
+        whenQueried().thenReturn(queryResponseMock);
+        helperGetLocalResourceOptions();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetLocalResourceOptionsEmptyResults() throws Exception {
+
+        whenQueried().thenReturn(queryResponseMock);
+        helperGetLocalResourceOptions();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetLocalResourceOptionsCatchesUnsupportedQueryException() throws Exception {
+
+        whenQueried().thenThrow(ResourceNotFoundException.class);
+        helperGetLocalResourceOptions();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetLocalResourceOptionsCatchesFederationException() throws Exception {
+
+        whenQueried().thenThrow(FederationException.class);
+        helperGetLocalResourceOptions();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetLocalResourceOptionsCatchesIllegalArgumentException() throws Exception {
+
+        whenQueried().thenThrow(IllegalArgumentException.class);
+        helperGetLocalResourceOptions();
+    }
+
+    protected void helperGetLocalResourceOptions() throws ResourceNotFoundException {
+        resourceOperations.getLocalResourceOptions(metacard.getId(), false);
```

Comment:
For these helperGet.... methods could they return the map so the calling function could validated the response or maybe leave them void and do the validation in the helper method themselves.